pulsebot/bugzilla.py

```diff
@@ -11,91 +11,59 @@
-        self._session.headers.update({"X-Bugzilla-API-Key": api_key})
-
-    def get_fields(self, bug, fields):
-        bug_url = "%s/rest/bug/%d?include_fields=%s" % (
-            self._server,
-            bug,
-            ",".join(fields),
-        )
-        logger.info(f"get_fields: {bug_url}")
-        try:
-            r = self._session.get(bug_url)
-            r.raise_for_status()
-            bug_data = r.json()
-        except Exception:
-            logger.exception(
-                f"Error occurred retrieving bug fields {bug_url} {r.status_code}"
-            )
-            raise BugzillaError()
-
+        self._headers = {
+            "User-Agent": "pulsebot",
+            "X-Bugzilla-API-Key": api_key,
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+
+    def _call(self, method, path, data=None):
+        data = data or {}
+        method = method or 'GET'
+        bug_url = f"{self._server}/{path}"
+        logger.info(f"{method}: {bug_url}")
+        if method.lower() == 'get':
+            r = requests.get(bug_url, headers=self._headers)
+        elif method.lower() == 'post':
+            r = requests.post(bug_url, data=data, headers=self._headers)
+        elif method.lower() == 'PUT':
+            r = requests.put(bug_url, data=data, headers=self._headers)
+        else:
+            raise BugzillaError(f"Unknown method: {method} {bug_url}")
+        if r.status_code != 200:
+            raise BugzillaError(f"Request Error: {method} {bug_url} {r.status_code} {r.reason}")
+        return r.json()
+
+    def _check_error(self, method, path, bug_data):
         if "error" in bug_data:
-            logger.error(
-                f"Error occurred retrieving bug fields {bug_url} {bug_data['error']}"
-            )
-            raise BugzillaError()
+            raise BugzillaError(f"Data Error: {method} {self._server}/{path} {bug_data['error']}")
 
+    def get_fields(self, bug, fields):
+        path = "rest/bug/%d?include_fields=%s" % (bug, ",".join(fields))
+        bug_data = self._call('GET', path)
+        self._check_error('GET', path, bug_data)
         return bug_data.get("bugs", [{}])[0]
 
     def get_comments(self, bug):
-        bug_url = "%s/rest/bug/%d/comment?include_fields=text" % (self._server, bug)
-
-        logger.info(f"get_comments: {bug_url}")
-        try:
-            r = self._session.get(bug_url)
-            r.raise_for_status()
-            bug_data = r.json()
-        except Exception:
-            logger.exception(
-                f"Error occurred retrieving comments for {bug_url} {r.status_code}"
-            )
-            raise BugzillaError()
-
-        if "error" in bug_data:
-            logger.error(
-                f"Error occurred retrieving comments for {bug_url} {bug_data['error']}"
-            )
-            raise BugzillaError()
-
+        path = f"rest/bug/{bug}/comment?include_fields=text"
```

Comment:
`include_fields=text` should be passed to the [`params`](https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls) argument.